CalibTracker/SiPixelESProducers/interface/SiPixel2DTemplateDBObjectESProducer.h

```diff
@@ -0,0 +1,33 @@
+#ifndef CalibTracker_SiPixelESProducers_SiPixel2DTemplateDBObjectESProducer_h
+#define CalibTracker_SiPixelESProducers_SiPixel2DTemplateDBObjectESProducer_h
+// -*- C++ -*-
+//
+// Package:    SiPixel2DTemplateDBObjectESProducer
+// Class:      SiPixel2DTemplateDBObjectESProducer
+// 
+/**\class SiPixel2DTemplateDBObjectESProducer SiPixel2DTemplateDBObjectESProducer.cc CalibTracker/SiPixelESProducers/plugin/SiPixel2DTemplateDBObjectESProducer.cc
+
+ Description: ESProducer for magnetic-field-dependent local reco templates
+
+ Implementation: Used inside the RecoLocalTracker/Records/TkPixelRecord to select the correct db for given magnetic field
+*/
+//
+// Original Author:  D.Fehling
+//         Created:  Tue Sep 29 14:49:31 CET 2009
+//
+//
+
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CondFormats/SiPixelObjects/interface/SiPixel2DTemplateDBObject.h"
+#include "CalibTracker/Records/interface/SiPixel2DTemplateDBObjectESProducerRcd.h"
+
+class SiPixel2DTemplateDBObjectESProducer : public edm::ESProducer  {
+
+public:
+
+	SiPixel2DTemplateDBObjectESProducer(const edm::ParameterSet& iConfig);
+        ~SiPixel2DTemplateDBObjectESProducer() override;
+	std::shared_ptr<SiPixel2DTemplateDBObject> produce(const SiPixel2DTemplateDBObjectESProducerRcd &);
```

Comment:
I believe the return value would be better as `unique_ptr`.

Reply:
Actually it is probably easier to keep this one as `shared_ptr` as the corresponding `SiPixelTemplateDBObjectESProducer` also still returns `shared_ptr`
https://github.com/cms-sw/cmssw/blob/bce87320443fede73263cb6e081cb3e97f2b037f/CalibTracker/SiPixelESProducers/plugins/SiPixelTemplateDBObjectESProducer.cc#L59

(this also links to Slava's comment https://github.com/cms-sw/cmssw/pull/22458#discussion_r176071551)